File: app/src/main/java/com/example/nijimac103/itunestracker/service/Artist.java
package com.example.nijimac103.itunestracker.service;

public class Artist {
    public String wrapperType;
    public String kind;
    public int artistId;
    public int collectionId;
    public int trackId;
    public String artistName;
    public String collectionName;
    public String trackName;
    public String collectionCensoredName;
    public String trackCensoredName;
    public String artistViewUrl;
    public String collectionViewUrl;
    public String trackViewUrl;
    public String previewUrl;
    public String artworkUrl30;
    public String artworkUrl60;
    public String artworkUrl100;
    public long collectionPrice;
    public long trackPrice;
    public String releaseDate;
    public String collectionExplicitness;
    public int discCount;
    public int discNumber;
    public int trackCount;
    public int trackNumber;
    public int trackTimeMillis;
    public String country;
    public String currency;
    public String primaryGenreName;

    public Artist() {
    }

    public Artist(String name) {
        this.artistName = name;
    }

}

File: app/src/main/java/jp/co/geisha/itunestracker/api/entity/Artist.kt
package jp.co.geisha.itunestracker.api.entity

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Artist(
        val artistName: String,
        val trackCensoredName: String,
        val previewUrl: String = "",
        val artworkUrl30: String = "",
        val artworkUrl60: String = "",
        val artworkUrl100: String = "",
        val primaryGenreName: String = ""
) {
    @JsonClass(generateAdapter = true)
    data class Result (
            @Json(name = "results")
            val data: List<Artist>
    )
}
